meteo/_utils.py

```diff
@@ -32,3 +36,95 @@ def normalize_longitude(ds: xr.Dataset, lon_name: str, to_360: bool = True) -> x
+
+
+def detect_name_in_ds(ds: xr.Dataset, candidates: list[str]):
+    for name in ds.coords:
+        if name.lower() in candidates:
+            return name
+
+    for name in ds.variables:
+        if name.lower() in candidates:
+            return name
+
+    raise ValueError(f"Could not auto-detect coord name in {candidates}")
+
+
+def detect_name(ds: xr.Dataset, var = "longitude") -> str:
+    if var == "longitude":
+        return detect_name_in_ds(ds, CANDIDATES_LON)
+    elif var == "latitude":
+        return detect_name_in_ds(ds, CANDIDATES_LAT)
+    else:
+        raise ValueError(f"No candidate associated with {var}")
+
+
+def detect_pad_width(ds: xr.Dataset, lon_name: str) -> int:
+    import numpy as np
+    ds = ds.sortby([lon_name])
+    lon = ds[lon_name].values
+
+    if ds[lon_name].ndim != 1:
+        raise ValueError(f"Longitude {lon_name} must be 1D for padding.")
+
+    left = lon[0]
+    right = lon[-1]
+    diffs = np.diff(lon)
+    left_step = diffs[0]
+    left_pad = int(np.ceil(abs(-180 - left)/left_step))
+    right_step = diffs[-1]
+    right_pad = int(np.ceil(abs(180 - right)/right_step))
+
+    pad_width = np.max([left_pad, right_pad])
+
+    if pad_width <= 0 or pad_width >= len(lon):
+        raise ValueError(f"Invalid auto-detected pad width: {pad_width}")
+
+    return pad_width
+
+
+def pad_lon(ds: xr.Dataset, pad_width: int, lon_name: str = "longitude") -> xr.Dataset:
+    # Pad the Dataset and Normalize longitude values of PAD.
+    # We use `wrap` mode in order to handle the Antimeridian
+    # Nevertheless it is necessary to "normalize" the longitudes due to the following issue.
+    # After padding the longitude values are like this:
+    #     179.7 179.9 -179.9 -177.7 .... 177.7 177.9 -177.9 -177.7
+    # This causes a problem if we convert to a pandas dataframe and then back to xarray:
+    #     df = ds.elevation.to_dataframe()
+    #     final = df.to_xarray()
+    # The `.to_xarray()` call reorders the index and messes things up. By normalizing
+    # the longitude values, i.e. by converting them to:
+    #     -180.3 -180.1 -179.9 -177.7 ... 177.7 177.9 180.1 180.3
+    # then no reordering happens and we can use normal .isel() to trim the adjusted dataset
+    import numpy as np
+    padded = ds.pad({lon_name: pad_width}, mode="wrap", keep_attrs=True)
+    lon = padded[lon_name].values
+    padded = padded.assign_coords({
+        lon_name: np.concatenate((
+            -180 - (180 - lon[:pad_width]),
+            lon[pad_width:-pad_width],
+            lon[-pad_width:] % 360,
+        )),
+    })
+    return padded
+
+
+def pad_lat(ds: xr.Dataset, lat_name: str = "latitude") -> xr.Dataset:
+    import numpy as np
+    ds = ds.sortby([lat_name])
+    new_lats = np.r_[-90, ds[lat_name], 90]
+    ds = ds.pad({lat_name: 1}, mode="median")
+    ds = ds.assign_coords(**{lat_name: new_lats})
+    return ds
+
+
+def write_file(ds: xr.Dataset, output_file: pathlib.Path, overwrite: bool
```

Comment:
both `.to_zarr()` and `.to_netcdf()` have a "mode" argument. a bit Different semantics, but it would probably make sense to try to use it for the overwrite functionality.